packages/web-media-box/src/hls-parser/tags/tagWithAttributesProcessors.ts

```diff
@@ -223,10 +223,24 @@ export class ExtXMedia extends TagWithAttributesProcessor {
       channels: tagAttributes[ExtXMedia.CHANNELS] ? tagAttributes[ExtXMedia.CHANNELS].split('/') : []
     };
 
-    if (!playlist.alternativeRenditions) {
-      playlist.alternativeRenditions = [];
+    const matchingGroup = playlist.renditionGroups?.find((group) => group.id === rendition.groupId && group.type === rendition.type);
+
+    if (matchingGroup) {
+      matchingGroup.renditions.push(rendition);
+      return;
+    }
+
+    // Create a new group for this rendition
+    const renditionGroup: RenditionGroup = {
+      id: rendition.groupId,
```

Comment:
Is it possible that a rendition does not have a `groupId`. If so, I'd guess we would want to skip the creation of the group. Otherwise, I think this is great!